querydsl-core/src/main/java/com/querydsl/core/support/CollectionAnyVisitor.java

```diff
@@ -110,8 +113,9 @@ protected Predicate exists(Context c, Predicate condition) {
             Path<?> parent = (Path<?>) expr.getMetadata().getParent().accept(this, context);
             expr = new PathImpl<Object>(expr.getType(), PathMetadataFactory.forCollectionAny(parent));
             EntityPath<?> replacement = new EntityPathBase<Object>(expr.getType(),
-                    ExpressionUtils.createRootVariable(expr));
+                    ExpressionUtils.createRootVariable(expr, replaced.size()));
             context.add(expr, replacement);
+            replaced.add(expr);
```

Comment:
Yes, shall we replace it with a simple counter? Putting the state into the visitor was easier than handling it via the context, because the context is new for each invocation.